subprojects/ide/src/integTest/groovy/org/gradle/plugins/ide/AbstractSourcesAndJavadocJarsIntegrationTest.groovy

```diff
@@ -294,6 +295,53 @@ dependencies {
         ideFileContainsEntry("module-1.0.jar", "module-1.0-sources.jar", "module-1.0-javadoc.jar")
     }
 
+    @ToBeFixedForInstantExecution
+    def "sources for gradleApi() are resolved and attached when -all distribution is used"() {
+        given:
+        requireGradleDistribution()
+        TestFile sourcesDir = distribution.gradleHomeDir.createDir("src")
+        sourcesDir.createFile("org/gradle/Test.java").writelns("package org.gradle;", "public class Test {}")
+
+        buildScript """
+apply plugin: "java"
+apply plugin: "idea"
+apply plugin: "eclipse"
+
+dependencies {
+    implementation gradleApi()
+}
+"""
+        when:
+        succeeds ideTask
+
+        then:
+        ideFileContainsGradleApiWithSources("gradle-api", sourcesDir.getPath())
+    }
+
+    @ToBeFixedForInstantExecution
+    def "sources for gradleTestKit() are resolved and attached when -all distribution is used"() {
+        given:
+        requireGradleDistribution()
+        TestFile sourcesDir = distribution.gradleHomeDir.createDir("src")
+        sourcesDir.createFile("org/gradle/Test.java").writelns("package org.gradle;", "public class Test {}")
+
+        buildScript """
+apply plugin: "java"
+apply plugin: "idea"
+apply plugin: "eclipse"
+
+dependencies {
+    implementation gradleTestKit()
```

Comment:
❓ What happen when `gradleApi()` and `gradleTestKit()` are used together? Will there be two source entries added? I'm sure it's not a big deal, we just need to make sure it's not a side effect but the intended behavior.

Reply:
A source entry will be added for test-kit and gradle-api jars - this is intended. I will add the assertion.